src/benchmark_api_internal.cc

```diff
@@ -114,5 +130,55 @@ void BenchmarkInstance::Teardown() const {
     teardown_(st);
   }
 }
+
+const std::array<Base10HumanReadableFormat, 5>
+    Base10HumanReadableFormat::ranges = {
+        Base10HumanReadableFormat{1, ""},             // from 0 to 10^3-1
+        Base10HumanReadableFormat{1000, "k"},         // from 10^3 to 10^6-1
+        Base10HumanReadableFormat{1000000, "m"},      // from 10^6 to 10^9-1
+        Base10HumanReadableFormat{1000000000, "bn"},  // from 10^9 to 10^12-1
+        Base10HumanReadableFormat{1000000000000, ""}  // above show as full
+};
+
+std::string Base10HumanReadableFormat::get(const int64_t& arg) {
```

Comment:
what's wrong with https://github.com/google/benchmark/blob/main/src/string_util.cc#L121 ?

Reply:
Oh I did not realise there is already a function for that. I will try to adapt this method such we can be used it for base10 (and have the correct abbreviations)